quill-core/src/test/scala/io/getquill/norm/FlattenOptionOperationSpec.scala

```diff
@@ -24,7 +25,31 @@ class FlattenOptionOperationSpec extends Spec { // hello
         o.getOrElse(1)
       }
       new FlattenOptionOperation(AnsiConcat, TraceConfig.Empty)(q.ast.body: Ast) mustEqual
-        If(BinaryOperation(Ident("o"), EqualityOperator.`_!=`, NullValue), Ident("o"), Constant.auto(1))
+        IfExist(o, o, c1)
+    }
+    "orElse" - {
+      "regular operation" in {
+        val q = quote { (o: Option[Int]) =>
+          o.orElse(Option(1))
+        }
+        new FlattenOptionOperation(AnsiConcat, TraceConfig.Empty)(q.ast.body: Ast) mustEqual
+          IfExist(o, o, c1)
+      }
+      "with forall" in {
+        val q = quote { (o: Option[Int]) =>
+          o.orElse(Option(1)).forall(_ == 2)
+        }
+        new FlattenOptionOperation(AnsiConcat, TraceConfig.Empty)(q.ast.body: Ast) mustEqual
+          ((o +==+ c2) +||+ (IsNullCheck(o) +&&+ (c1 +==+ c2))
+            +||+ (IsNullCheck(o) +&&+ IsNullCheck(c1)))
```

Comment:
First time I see this notation, I think I kind of understand what it means but can you maybe give me a quick explanation of what it does/represents, please?